src/serializer/serializer.js

```diff
@@ -1252,8 +1253,10 @@ export class Serializer {
           let serializedBinding: SerializedBinding = serializedBindings[name];
           if (serializedBinding.modified && !serializedBinding.referentialized) {
             let serializedBindingId = t.identifier(this.referentializedNameGenerator.generate(name));
+            let serializedValue = serializedBinding.serializedValue;
+            invariant(serializedValue);
```

Comment:
It's true because the serializer has two distinct phases: first it walks the help, serializing values, noting residual functions and establishing the bindings, but not actually serializing residual functions yet. At that point, the bindings are all initialized. Only then this code kicks in which actually writes out the residual functions, and that can now rely on all serialized values and bindings being initialized.